From 1627f1046515e93f7acb34fbfededecf9e1b16a7 Mon Sep 17 00:00:00 2001
From: Adam Duskett <Aduskett@gmail.com>
Date: Sun, 29 Mar 2020 16:28:06 -0700
Subject: [PATCH] improvement: build: add an option to disable building
 documentation

For embedded applications such as Buildroot or Yocto, the man pages may not be
desired or even capable of being built.

On line 45 of configure.ac there is the line:
JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl], [DocBook XSL Stylesheets])

There are three issues with this:
  - It requires building the xml-catalog package.
  - It automatically defaults to the host systems xml-catalog
    instead of the cross environments.
  - It isn't necessary to have a functioning firewalld.

Create a new option: --disable-docs. By default, build the documentation, but
if a user passes --disable-docs documentation will no longer be built as
desired.

Upstream-Status: Merged
See: https://github.com/firewalld/firewalld/pull/596

Signed-off-by: Adam Duskett <aduskett@gmail.com>
---
 configure.ac             | 7 ++++++-
 doc/man/man1/Makefile.am | 2 ++
 doc/man/man5/Makefile.am | 2 ++
 doc/xml/Makefile.am      | 2 ++
 4 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 0758c69d..84f48037 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,12 @@ GLIB_GSETTINGS
 
 #############################################################
 
-JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl], [DocBook XSL Stylesheets])
+AC_ARG_ENABLE([docs],
+    [AS_HELP_STRING([--disable-docs], [Disable building documentation])])
+AM_CONDITIONAL([ENABLE_DOCS], [test x$enable_docs != xno])
+AM_COND_IF([ENABLE_DOCS], [
+    JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl], [DocBook XSL Stylesheets])
+])
 
 #############################################################
 
diff --git a/doc/man/man1/Makefile.am b/doc/man/man1/Makefile.am
index 23343490..78c42fdc 100644
--- a/doc/man/man1/Makefile.am
+++ b/doc/man/man1/Makefile.am
@@ -1,3 +1,5 @@
+if ENABLE_DOCS
 EXTRA_DIST = $(man_MANS)
 
 man_MANS = firewall*.1
+endif
diff --git a/doc/man/man5/Makefile.am b/doc/man/man5/Makefile.am
index 6ba37a9b..2ed35461 100644
--- a/doc/man/man5/Makefile.am
+++ b/doc/man/man5/Makefile.am
@@ -1,3 +1,5 @@
+if ENABLE_DOCS
 EXTRA_DIST = $(man_MANS)
 
 man_MANS = firewall*.5
+endif
diff --git a/doc/xml/Makefile.am b/doc/xml/Makefile.am
index d0313e3e..44a32528 100644
--- a/doc/xml/Makefile.am
+++ b/doc/xml/Makefile.am
@@ -1,5 +1,6 @@
 XSLTPROC = xsltproc
 
+if ENABLE_DOCS
 EXTRA_DIST = $(HTMLS:../html/%.html=%.xml) \
 	authors.xml notes.xml seealso.xml errorcodes.xml \
 	transform-man.xsl.in transform-html.xsl.in \
@@ -27,6 +28,7 @@ man5_MANS = \
 	../man/man5/firewalld.service.5 \
 	../man/man5/firewalld.zone.5 \
 	../man/man5/firewalld.zones.5
+endif
 
 CLEAN_FILES = *~ errorcodes.xml
 DISTCLEANFILES = $(man_MANS) $(HTMLS) transform-*.xsl \
